package/setup.py

```diff
@@ -253,7 +253,19 @@ def using_clang():
     compiler = new_compiler()
     customize_compiler(compiler)
     compiler_ver = getoutput("{0} -v".format(compiler.compiler[0]))
-    return 'clang' in compiler_ver
+    if 'Spack GCC' in compiler_ver:
```

Comment:
So I have no issues with the setup file as-is, but I do wonder if this is opening a pandora's box in supporting the peculiarities for various build tools (for example, does EasyBuild also need handling differently?).

Is this the same way numpy/scipy is handling compiler identification @tylerjereddy ?

Reply:
No, SciPy uses `scipy/_build_utils/compiler_helper.py`, which is a module dedicated to compiler handling. As discussed in the matching issue, they implement a small `try_compile()` function to only allow flags to get appended to the compile line if the compiler genuinely supports them.

We could redesign to use something like that--that may be biting off a bit more than I was hoping to chew here, though we could probably even copy-paste a decent bit of the code.

Here's an example of how it works:

```python
    if sys.platform == 'darwin':
        # Set min macOS version
        min_macos_flag = '-mmacosx-version-min=10.9'
        if has_flag(cc, min_macos_flag):
            args.append(min_macos_flag)
            ext.extra_link_args.append(min_macos_flag)
```

That `has_flag()` function checks that the compile flag works "for real" with the compiler:

```python
def has_flag(compiler, flag, ext=None):
    """Returns True if the compiler supports the given flag"""
    return try_compile(compiler, flags=[flag], ext=ext)
```

Reply:
Is this starting to re-implement what automake and cmake build systems do?